spawn_worker/src/unix/fork.rs

```diff
@@ -16,7 +18,9 @@ pub enum Fork {
 /// Existing state of the process must allow safe forking, e.g. no background threads should be running
 /// as any locks held by these threads will be locked forever
 ///
-pub unsafe fn fork() -> Result<Fork, std::io::Error> {
+/// When forking a multithreaded application, no code should allocate or access other potentially locked resources
+/// until call to exec is executed
+pub(crate) unsafe fn fork() -> Result<Fork, std::io::Error> {
     let res = libc::fork();
     match res {
         -1 => Err(std::io::Error::last_os_error()),
```

Comment:
Pretty sure this is strictly equivalent to [nix::fork](https://docs.rs/nix/latest/nix/unistd/fn.fork.html)

Reply:
You're right :facepalm:  
I'll promise to fix this in other pr :pinky_swear: